### PR TITLE
Fix mypy errors for pandas\tests\* #28926 (test_algos.py)

### PR DESCRIPTION
**Pull request by lrm25**
- [x] xref #28926
- [x] tests added / passed (verified that tests still pass)
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [N/A] whatsnew entry

### CODE REVIEW

File: pandas/tests/test_algos.py [hunk 1]
```diff
@@ -767,7 +767,7 @@ def test_same_object_is_in(self):
         # with similar behavior, then we at least should
         # fall back to usual python's behavior: "a in [a] == True"
         class LikeNan:
-            def __eq__(self):
+            def __eq__(self, other: object):
```
Comment (simonjayhawkins) [inline pandas/tests/test_algos.py]: I _think_ we only need to correct the signature to resolve the error at this stage. @WillAyd 

```suggestion
            def __eq__(self, other):
```

**Comment by simonjayhawkins** [comment]
Thanks @lrm25 for the PR.

**Review by simonjayhawkins — CHANGES_REQUESTED** [review]
@lrm25 can you remove the test_algos section from setup.cfg

**Review by simonjayhawkins — APPROVED** [review]
@lrm25 lgtm @WillAyd

**Comment by WillAyd** [comment]
Nice find. Not sure this was even testing what was intended previously, so definitely speaks to annotations being useful for tests

**Comment by WillAyd** [comment]
Thanks!

**Pull request merged**